test/test_allocation_domain.cpp

```diff
@@ -1195,4 +1195,42 @@ TEST_F(NVFuserTest, AllocationDomainContiguityForExplicitBroadcast) {
   testValidate(fusion, outputs, {t0}, {t1}, __LINE__, __FILE__);
 }
 
+TEST_F(AllocationDomainTest, VectorizeOverlappingTensor) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(3);
+  fusion.addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = set(tv1);
+  fusion.addOutput(tv2);
+
+  // According to the stride order below, the allocation domain should be the
+  // same as the root domain. However, here we intentionally make the allocation
+  // domain [axis(1), axis(0), axis(2)] because doing so allows us to vectorize
+  // by 4.
+  tv0->setAllocationDomain(
+      {tv0->axis(1), tv0->axis(0), tv0->axis(2)}, {false, true, true});
+
+  for (auto tv : {tv2, tv1}) {
+    tv->reorder({{0, 1}});
+    tv->merge(0);
+    tv->merge(0);
+    tv->split(0, 4);
+    tv->axis(0)->parallelize(ParallelType::TIDx);
+  }
+  tv1->axis(1)->parallelize(ParallelType::Vectorize);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 =
+      at::randn({4 * 5 * 7}, options).as_strided({4, 5, 7}, {7, 4, 1});
```

Comment:
Just to confirm my understanding, the `4` in `{7,4,1}` can be changed to any value between 1 and 7, exclusively, without affecting the behavior of the test. Correct?

Reply:
Yes, except that it must be compatible with the vectorization factor. If you still want to vectorize 4, then it must be 4 or 0. If you want to vectorize 2, then it can be 0, 2, 4, 6

Reply:
> If you still want to vectorize 4, then it must be 4 or 0.

How so? ({4,5,7},{7,x,1}) can be viewed as ({5,28},{x,1}). Because 28 is a multiple of 4, we can always vectorize the tensor by 4 regardless of `x`. Correct?

Reply:
Assume the tensor's base address is 0, and x is 1, then the slice `[:, 1, :]`'s base address will be 1. This is not aligned to 4, and you will get a "misaligned address" error if you try to vectorize 4.

Reply:
For posterity, can you add a comment explaining `stride=[1]` has to be a multiple of 4 for alignment?